### PR TITLE
[NBS Docs] for catalog-backend-module-incremental-ingestion plugin

### PR DESCRIPTION
**Pull request by zeshanziya**
## Hey, I just made a Pull Request!

Add new backend setup doc for plugin https://github.com/backstage/backstage/tree/master/plugins/catalog-backend-module-incremental-ingestion

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jhaals — COMMENTED** [review]
So great to have these new docs! We are really close to start migrating and asking for contributions for deprecating `createRouter` exports and fully transition to the new backend system. Given that could we make sure that your additions are listed before the old backend in the instructions to encourage users to use the new option.

**Comment by zeshanziya** [comment]
Thanks @jhaals. I have moved old backend doc at bottom.

**Comment by Parsifal-M** [comment]
Nice one! @jhaals Just wondering, and also maybe @ahhhndre / @sennyeya in terms of these NBS docs are we following this pattern? e.g. https://github.com/backstage/backstage/pull/25805 

Where we make an `--old` for the old backend?

WDYT?

Thanks! 🙏

**Comment by awanlin** [comment]
@Parsifal-M as this is in a `README.md` it's hard to apply that pattern here. I think in this case merging this PR as is makes sense but then adding it somewhere in the Software Catalog section of the docs would be great to raise awareness.

File: plugins/catalog-backend-module-incremental-ingestion/README.md [hunk 1]
```diff
@@ -307,9 +275,136 @@ export class MyIncrementalEntityProvider implements IncrementalEntityProvider<Cu
 
 Now that you have your new Incremental Entity Provider, we can connect it to the catalog.
 
-## Adding an Incremental Entity Provider to the catalog
+## Adding an Incremental Entity Provider to the catalog (New Backend)
```
Comment (awanlin) [inline plugins/catalog-backend-module-incremental-ingestion/README.md:278]:
```suggestion
## Adding an Incremental Entity Provider to the catalog
```

File: plugins/catalog-backend-module-incremental-ingestion/README.md [hunk 2]
```diff
@@ -307,9 +275,136 @@ export class MyIncrementalEntityProvider implements IncrementalEntityProvider<Cu
...
+  await processingEngine.start();
+
+  return router;
+}
+```
+
+## Adding an Incremental Entity Provider to the catalog (Old Backend)
 
-We'll assume you followed the <a href="#installation">Installation</a> instructions. After you create your `incrementalBuilder`, you can instantiate your Entity Provider and pass it to the `addIncrementalEntityProvider` method.
+We'll assume you followed the <a href="#installation-old-backend">Installation</a> instructions. After you create your `incrementalBuilder`, you can instantiate your Entity Provider and pass it to the `addIncrementalEntityProvider` method.
```
Comment (awanlin) [inline plugins/catalog-backend-module-incremental-ingestion/README.md:407]:
```suggestion
We'll assume you followed the [Installation](#installation-old-backend) instructions. After you create your `incrementalBuilder`, you can instantiate your Entity Provider and pass it to the `addIncrementalEntityProvider` method.
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @zeshanziya, left some comments, most are pretty simple.

As mentioned already if you are up for it in a follow up PR it would be awesome to have this as part of the main docs to help raise visibility to this option for those who need it.

**Comment by zeshanziya** [comment]
Thanks, @awanlin! I'll work on the suggested changes. Regarding the follow-up PR, are you suggesting that this document be added to the main documentation? If so, where should it be linked, perhaps under the [External Integrations](https://backstage.io/docs/features/software-catalog/external-integrations) page as an "Incremental Entity Providers" section?

**Comment by awanlin** [comment]
Yes, that sounds like a good place for it to go @zeshanziya 👍

**Comment by zeshanziya** [comment]
Hey @awanlin, I have made the suggested changes. Please review. Thank you.

**Comment by jhaals** [comment]
Glad to see docs improvements! I just want to point out that we plan to completely remove the old backend system this year so I don't think we should much time on having docs for both new and old system, I rather have us just support the new system as we are currently working on deprecating `createRouter` and similar exports across all backend plugins. Meta issue will follow with some more context and how to update each plugin accordingly.

**Comment by awanlin** [comment]
I'll re-review once the changes @jhaals requested are made @zeshanziya 👍

**Comment by zeshanziya** [comment]
Thanks @jhaals @awanlin. As suggested, I have removed old backend related docs.

File: plugins/catalog-backend-module-incremental-ingestion/README.md [hunk 3]
```diff
@@ -42,57 +42,25 @@ The Incremental Entity Provider backend is designed for data sources that provid
...
-const builder = CatalogBuilder.create(env);
-// incremental builder receives builder because it'll register
-// incremental entity providers with the builder
-const incrementalBuilder = await IncrementalCatalogBuilder.create(env, builder);
-```
+2. Add the following code to the `packages/backend/src/index.ts` file:
 
-3. After building the regular `CatalogBuilder`, build the incremental builder:
+```diff
+++import { catalogModuleCustomIncrementalIngestionProvider } from './extensions/catalogCustomIncrementalIngestion';
```
Comment (jhaals) [inline plugins/catalog-backend-module-incremental-ingestion/README.md:49]: There are a few places with double plus signs, one is enough.

```suggestion
+ import { catalogModuleCustomIncrementalIngestionProvider } from './extensions/catalogCustomIncrementalIngestion';
```

**Review by jhaals — COMMENTED** [review]
Looking good to me, just some minor things to correct with the diffs :)

File: plugins/catalog-backend-module-incremental-ingestion/README.md (hunk 3 above)
Comment (zeshanziya) [reply plugins/catalog-backend-module-incremental-ingestion/README.md]: Thanks @jhaals. Double plus signs removed.

**Review by jhaals — APPROVED** [review]
Thanks! Looks good to me, letting @awanlin take a final look :)

**Review by awanlin — APPROVED** [review]

**Pull request merged**